File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        
        ListNode res = new ListNode(0);
        ListNode curr = res;
        res.next = head;
        
        while(curr != null && curr.next != null){
            
            if(curr.next.val == val){
                ListNode next2 = curr.next.next;
                curr.next = next2;
            }else {
                curr = curr.next;
            }
        }
        
        return res.next;
        
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        
        ListNode* prev = nullptr;
        ListNode* curr = head;
        
        while(curr != nullptr){
            
            if(curr->val == val){
                
                if(prev == nullptr){
                    head = curr->next;
                }else {
                    prev->next = curr->next;
                }
                
                ListNode* temp = curr;
                curr = curr->next;
                delete temp;
                
            }else {
                prev = curr;
                curr = curr->next;
            }
            
        }
        
        return head;
        
    }
};